Tasks/RunPowerShellOnTargetMachines/RunPowerShellOnTargetMachines.ps1

```diff
@@ -55,50 +56,86 @@ $credential = New-Object 'System.Net.NetworkCredential' -ArgumentList $machineUs
         Write-Verbose "Initiating deployment on $fqdn" -Verbose
 
-        $deploymentResponse = Invoke-PsOnRemote -MachineDnsName $fqdn -ScriptPath $scriptPath -WinRMPort $port -Credential $credential -ScriptArguments $scriptArguments -InitializationScriptPath $initializationScriptPath -ApplicationPath $applicationPath �SkipCACheck -UseHttp
+        $deploymentResponse = Invoke-PsOnRemote -MachineDnsName $fqdn -ScriptPath $scriptPath -WinRMPort $port -Credential $credential -ScriptArguments $scriptArguments -InitializationScriptPath $initializationScriptPath �SkipCACheck -UseHttp
 
         $log = "Deployment Logs : " + $deploymentResponse.DeploymentLog + "`nService Logs : " + $deploymentResponse.ServiceLog;
 
         Output-ResponseLogs -operationName "deployment" -fqdn $fqdn -deploymentResponse $deploymentResponse
 
         $response = $deploymentResponse
-    }
+        $status = $response.Status
+        Write-Output "Deployment Status for machine $fdqn : $status"
 
-    $logs = New-Object 'System.Collections.Generic.List[System.Object]'         
-    $resourceOperationLog = New-OperationLog -Content $log
-    $logs.Add($resourceOperationLog)
+        $logs = New-Object 'System.Collections.Generic.List[System.Object]'         
+        $resourceOperationLog = New-OperationLog -Content $log
+        $logs.Add($resourceOperationLog)
 
-    Complete-ResourceOperation -EnvironmentName $environmentName -EnvironmentOperationId $envOperationId -ResourceOperationId $resOperationId -Status $response.Status -ErrorMessage $response.Error -Logs $logs -ErrorAction Stop -Connection $connection
+        Complete-ResourceOperation -EnvironmentName $environmentName -EnvironmentOperationId $envOperationId -ResourceOperationId $resOperationId -Status $response.Status -ErrorMessage $response.Error -Logs $logs -ErrorAction Stop -Connection $connection
     
-    if ($response.Status -ne "Passed")
+        if ($response.Status -ne "Passed")
+        {
+            Complete-EnvironmentOperation -EnvironmentName $environmentName -EnvironmentOperationId $envOperationId -Status "Failed" -ErrorAction Stop -Connection $connection
+
+            throw $response.Error;
+        }
+    }
+}
+else
+{
+    $Jobs = New-Object "System.Collections.Generic.Dictionary``2[int, string]"
+
+    foreach($resource in $resources)
     {
-        Complete-EnvironmentOperation -EnvironmentName $environmentName -EnvironmentOperationId $envOperationId -Status "Failed" -ErrorAction Stop -Connection $connection
+        $machine = $resource.Name
+	
+        Write-Output "Deployment will be Started for - $machine"
 
-        throw $response.Error;
+        $job = Start-Job -ScriptBlock $RunPowershellJob -ArgumentList $environmentName, $envOperationId, $machine, $scriptPath, $port, $scriptArguments, $initializationScriptPath, $credential, $connection
+
+        $Jobs.Add($job.Id, $machine)
+    
+        Write-Output "Deployment Started for  - $machine"
+    }
+    While (Get-Job)
```

Comment:
I can see that the logic of this get-job is also shared, and it should be put in a shared module as well.